include/gridtools/fn/cartesian2.hpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2021, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#pragma once
+
+#include "../common/tuple_util.hpp"
+#include "../sid/concept.hpp"
+#include "./backend2/common.hpp"
+#include "./executor.hpp"
+
+namespace gridtools::fn {
+    namespace cartesian::dim {
+        using i = integral_constant<int, 0>;
+        using j = integral_constant<int, 1>;
+        using k = integral_constant<int, 2>;
+    } // namespace cartesian::dim
+
+    namespace cartesian_impl_ {
+        namespace dim = cartesian::dim;
+        using backend::data_type;
+        using backend::data_type_from_sid;
+
+        using domain_t = hymap::keys<dim::i, dim::j, dim::k>::values<int, int, int>;
+
+        inline domain_t cartesian_domain(int i, int j, int k) { return {i, j, k}; }
```

Comment:
Was it a conscious decision to remove offset from domain? Now we would need to shift origin of all input fields.

Reply:
Added offsets now.